src/Catalyst.Core.Modules.Kvm/DeltaExecutor.cs

```diff
@@ -0,0 +1,411 @@
+                env.ExecutingAccount = recipient;
+                env.CurrentBlock = stateUpdate;
+                env.GasPrice = gasPrice;
+                env.InputData = data ?? new byte[0];
+                env.CodeInfo = isPrecompile ? new CodeInfo(recipient) : machineCode == null ? _virtualMachine.GetCachedCodeInfo(recipient) : new CodeInfo(machineCode);
+                env.Originator = sender;
+
+                ExecutionType executionType = entry.IsValidDeploymentEntry ? ExecutionType.Create : ExecutionType.Call;
+                using (VmState state = new VmState(unspentGas, env, executionType, isPrecompile, true, false))
+                {
+                    substate = _virtualMachine.Run(state, txTracer);
+                    unspentGas = state.GasAvailable;
+                }
+
+                if (substate.ShouldRevert || substate.IsError)
+                {
+                    if (_logger.IsEnabled(LogEventLevel.Verbose)) _logger.Verbose("Restoring state from before transaction");
+                    _stateProvider.Restore(stateSnapshot);
+                    _storageProvider.Restore(storageSnapshot);
+                }
+                else
+                {
+                    // tks: there is similar code fo contract creation from init and from CREATE
+                    // this may lead to inconsistencies (however it is tested extensively in blockchain tests)
+                    if (entry.IsValidDeploymentEntry)
+                    {
+                        long codeDepositGasCost = CodeDepositHandler.CalculateCost(substate.Output.Length, spec);
+                        if (unspentGas < codeDepositGasCost && spec.IsEip2Enabled)
+                        {
+                            throw new OutOfGasException();
+                        }
+
+                        if (unspentGas >= codeDepositGasCost)
+                        {
+                            Keccak codeHash = _stateProvider.UpdateCode(substate.Output);
+                            _stateProvider.UpdateCodeHash(recipient, codeHash, spec);
+                            unspentGas -= codeDepositGasCost;
+                        }
+                    }
+
+                    foreach (Address toBeDestroyed in substate.DestroyList)
+                    {
+                        if (_logger.IsEnabled(LogEventLevel.Verbose)) _logger.Verbose($"Destroying account {toBeDestroyed}");
+                        _stateProvider.DeleteAccount(toBeDestroyed);
+                    }
+
+                    statusCode = StatusCode.Success;
+                }
+
+                spentGas = Refund(gasLimit, unspentGas, substate, sender, gasPrice, spec);
+            }
+            catch (Exception ex) when (ex is EvmException || ex is OverflowException) // TODO: OverflowException? still needed? hope not
+            {
+                if (_logger.IsEnabled(LogEventLevel.Verbose)) _logger.Verbose($"EVM EXCEPTION: {ex.GetType().Name}");
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Add curly braces around the nested statement(s) in this 'if' block.](https://app.codacy.com/app/catalyst-network/Catalyst.Node/pullRequest?prid=4350015)